plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -212,6 +222,8 @@ processTerm = handleTerm <=< traverseOf termSubtypes applyTypeSubstitution where
         -- Already processed term, just rename and put it in, don't do any
         -- further optimization here.
         Done t -> PLC.rename t
+        -- We still need to process the suspended term
+        Susp t -> PLC.rename t >>= processTerm
```

Comment:
So I read it again, and I conclude that I don't understand the deal with Susp. The argument is that you want the RHS of the binding to be processed only once. But if we processed it when we see the let-binding... it would _still_ only be processed once. The only way it would get processed twice is if we simplified _after_ we substituted in a variable, which I'm pretty sure we don't.

Reply:
Also this version of `Susp` is wrong: it should capture the substitution at the point we saw the binding. This will use the substitution at the usage site, which is potentially wrong in the presence of shadowing (maybe this could be the source of your bug?).

Reply:
Aha! Looking at section 7.1, I see that they *do* immediately call `simplExpr` on when they've substituted in the RHS of a variable. I think we're just biting the bullet that we might run the whole simplifier more times, and so we don't care. That maybe saves us from having to do the whole `Susp` dance.

Reply:
Yes, in the end I decided to just have a `Done` case for now.